### PR TITLE
feat: add additional geodesic calculations

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

**Review by Derollez — APPROVED** [review]
LGTM, just wondering if `Forward` and `IntermediateBetween` are clear/explicit enough from a user perspective? If you think it is, let's go

**Comment by vishwa2710** [comment]
@Derollez 

> LGTM, just wondering if `Forward` and `IntermediateBetween` are clear/explicit enough from a user perspective? If you think it is, let's go

Yeah I totally agree, I definitely did not love Forward, it's used in the geospatial community as it solves the "Forward" geodesic problem, but I'm open to better naming! cc: @kyle-cochran @apaletta3 @phc1990

File: src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp [hunk 1]
```diff
@@ -251,6 +286,117 @@ Length LLA::DistanceBetween(
...
+
+Array<LLA> LLA::IntermediateLLAs(
+    const LLA& aFirstLLA,
+    const LLA& aSecondLLA,
+    const Size& aNumberOfPoints,
+    const Length& anEllipsoidEquatorialRadius,
+    const Real& anEllipsoidFlattening
+)
+{
+    Array<LLA> intermediateLLAs;
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp:375]:
```suggestion
    Array<LLA> intermediateLLAs = Array<LLA>::Empty();
```

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp [hunk 2]
```diff
@@ -61,6 +67,62 @@ class LLA
...
+
+    /// @brief                  Calculate a point between this LLA coordinate and another LLA coordinate
+    ///
+    /// @param                  [in] aLLA A LLA coordinate
+    /// @param                  [in] aRatio A ratio
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 A point between the two LLA coordinates
+
+    LLA calculateIntermediateTo(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:90]: For the naming, what about 
`calculateIntermediateLLATo`

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp [hunk 3]
```diff
@@ -61,6 +67,62 @@ class LLA
...
+
+    /// @brief                  Propagate this LLA coordinate in provided direction and distance
+    ///
+    /// @param                  [in] aDirection An azimuth angle
+    /// @param                  [in] aDistance A distance
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 Propagated LLA coordinate
+
+    LLA calculateForward(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:105]: For the naming, what about 
`calculateForwardLLA`

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp [hunk 4]
```diff
@@ -92,6 +154,72 @@ class LLA
...
+    /// @brief                  Calculate a point between two LLA coordinates
+    ///
+    /// @param                  [in] aFirstLLA A first LLA coordinate
+    /// @param                  [in] aSecondLLA A second LLA coordinate
+    /// @param                  [in] aRatio A ratio
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 A point between the two LLA coordinates
+
+    static LLA IntermediateBetween(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:181]: For the naming, what about
`IntermediateLLABetween`

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp [hunk 5]
```diff
@@ -92,6 +154,72 @@ class LLA
...
+    /// @brief                  Propagate LLA coordinates in provided direction and distance
+    ///
+    /// @param                  [in] aLLA A LLA coordinate
+    /// @param                  [in] aDirection An azimuth angle
+    /// @param                  [in] aDistance A distance
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 Propagated LLA coordinate
+
+    static LLA Forward(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:198]: For the naming, what about
`ForwardLLA`

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp [hunk 6]
```diff
@@ -92,6 +154,72 @@ class LLA
...
+    /// @brief                  Generate equidistant intermediate LLAs between two LLA coordinates.
+    ///
+    /// @param                  [in] aFirstLLA A first LLA coordinate
+    /// @param                  [in] aSecondLLA A second LLA coordinate
+    /// @param                  [in] aNumberOfPoints A number of points
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 intermediate equidistant LLAs between the two LLA coordinates
+
+    static Array<LLA> IntermediateLLAs(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:215]: For the naming, what about 
`IntermediateLLAsBetween`

**Review by apaletta3 — APPROVED** [review]
Nice stuff! Left some naming suggestions

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 2 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:90]: Hmmm I think then the others should be named `calculateDistanceToLLA` and `calculateAzimuthAngleBetweenLLA`.
IMO the LLA part is fairly obvious, especially from the function arguments.

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 3 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:105]: Same as above.

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 6 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:215]: Ooooh. How about `Linspace`. That's a popularly used function in other libraries that most people are familiar with.

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 5 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:198]: Same as above, IMO the LLA part is obvious based on the function arguments and class.

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 4 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:181]: Same as above, IMO the LLA part is obvious based on the function arguments and class.

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 6 above)
Comment (apaletta3) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:215]: Yeah I like that, good idea. I still think you should be consistent with the naming between the regular methods and static methods of each kind, cause there are a few discrepancies in their naming

File: include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp (hunk 2 above)
Comment (apaletta3) [reply include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp:90]: sure

**Pull request merged**